lib/src/Uncertainty/Algorithm/MetaModel/FunctionalChaos/FunctionalChaosResult.cxx

```diff
@@ -273,6 +279,131 @@ Function FunctionalChaosResult::getComposedMetaModel() const
   return composedMetaModel_;
 }
 
+/* Conditional expectation accessor */
+FunctionalChaosResult FunctionalChaosResult::getConditionalExpectation(const Indices & conditioningIndices) const
+{
+  // Get marginal input sample and distribution
+  const UnsignedInteger inputDimension = inputSample_.getDimension();
+  conditioningIndices.check(inputDimension);
+  const Sample inputSampleMarginal(inputSample_.getMarginal(conditioningIndices));
+  const Distribution inputDistributionMarginal(distribution_.getMarginal(conditioningIndices));
+
+  // Restrict the orthogonal basis
+  // TODO: create a new OrthogonalBasis.getMarginal(activeIndices)
+  const String basicClassName(orthogonalBasis_.getImplementation()->getClassName());
+  if (basicClassName != "OrthogonalProductPolynomialFactory")
```

Comment:
The actual test should be to check if the measure associated to the basis has an independent copula

Reply:
We need to make sure that the code works also for a `OrthogonalProductFunctionFactory`. 
Since there is a dynamic cast, there is no need to check the name of the class. 
We should change the code so that `OrthogonalProductPolynomialFactory` derived from `OrthogonalProductFunctionFactory`. The code in `FunctionalChaosResult` requires only a `OrthogonalProductFunctionFactory`.

Reply:
This can be simplified after #2743.